pkg/provider/identity.go

```diff
@@ -6,27 +6,46 @@ package provider
 import (
 	"context"
 	"crypto"
-	"crypto/sha256"
 	"crypto/tls"
 	"crypto/x509"
+	"io/ioutil"
 	"net"
+	"os"
 
-	base58 "github.com/jbenet/go-base58"
+	"github.com/zeebo/errs"
 	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 
-	"storj.io/storj/pkg/dht"
 	"storj.io/storj/pkg/peertls"
+	"storj.io/storj/pkg/utils"
 )
 
+const (
+	IdentityLength = uint16(256)
+)
+
+var (
+	ErrDifficulty = errs.Class("difficulty error")
+)
+
+// CertificateAuthority represents the CA which is used to author and validate identities
+type CertificateAuthority struct {
+	// PrivateKey is the private key of the CA
+	PrivateKey crypto.PrivateKey
+	// Cert is the x509 certificate of the CA
+	Cert *x509.Certificate
+	// The ID is calculated from the CA cert.
+	ID nodeID
+}
+
 // PeerIdentity represents another peer on the network.
 type PeerIdentity struct {
 	// CA represents the peer's self-signed CA. The ID is taken from this cert.
-	CA *x509.Certificate
+	CA CertificateAuthority
```

Comment:
hmm, but a CertificateAuthority has a PrivateKey field. you definitely won't have that for a peer